<launch>

  <arg name="gui" default="true" />
  <arg name="gpu" default="-1" />
  <arg name="network" default="alexnet" />

  <node name="image_publisher"
        pkg="jsk_perception" type="image_publisher.py">
    <rosparam subst_value="true">
      file_name: $(find jsk_perception)/sample/image/leopard.jpg
      publish_info: false
      encoding: bgr8
    </rosparam>
  </node>

  <include file="$(find jsk_perception)/launch/imagenet_object_recognition.launch">
    <arg name="INPUT_IMAGE" value="/image_publisher/output" />
    <arg name="gpu" value="$(arg gpu)" />
    <arg name="network" value="$(arg network)" />
  </include>

  <node name="draw_classification_result"
        pkg="jsk_perception" type="draw_classification_result.py">
    <remap from="~input" to="object_recognition/output" />
    <remap from="~input/image" to="image_publisher/output" />
  </node>

  <group if="$(arg gui)">
    <node name="image_view0"
          pkg="image_view" type="image_view">
      <remap from="image" to="draw_classification_result/output" />
    </node>
  </group>

</launch>
